modules/account-lib/src/coin/dot/transactionBuilder.ts

```diff
@@ -1,57 +1,480 @@
+   */
+  genesisHash(genesisHash: string): this {
+    this._genesisHash = genesisHash;
+    return this;
+  }
+
+  /**
+   *
+   * The raw metadata of the chain in string format.
+   *
+   * @param {string} metadataRpc
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  metadataRpc(metadataRpc: string): this {
+    this._metadataRpc = metadataRpc;
+    return this;
+  }
+
+  /**
+   *
+   * The specVersion for transaction format.
+   *
+   * @param {number} specVersion
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  specVersion(specVersion: number): this {
+    this._specVersion = specVersion;
+    return this;
+  }
+
+  /**
+   *
+   * The chainName for transaction format.
+   *
+   * @param {number} chainName
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  chainName(chainName: string): this {
+    this._chainName = chainName;
+    return this;
+  }
+
+  private method(method: TxMethod): this {
+    this._method = method;
+    return this;
+  }
+
+  /**
+   * Set the network based on the configuration in the statics module
+   */
+  private staticsConfig(): void {
+    const networkConfig = this._coinConfig.network as DotNetwork;
+    this.specName(networkConfig.specName);
+    this.genesisHash(networkConfig.genesisHash);
+    this.specVersion(networkConfig.specVersion);
+    this.chainName(networkConfig.chainName);
+  }
+
+  /**
+   * Sets the
+   * specName,
+   * genesisHash - genesisHash of the chain,
+   * metadataRpc - The SCALE-encoded metadata for the runtime when submitted,
+   * specVersion - The current spec version for the runtime,
+   * chainName - chainName,
+   * registry,
+   * of the westened
+   *
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  westend(): this {
+    this.metadataRpc(westendMetadataRpc);
+    this.buildRegistry();
+    return this;
+  }
+
+  /**
+   * Sets the
+   * specName,
+   * genesisHash - genesisHash of the chain,
+   * metadataRpc - The SCALE-encoded metadata for the runtime when submitted,
+   * specVersion - The current spec version for the runtime,
+   * chainName - chainName,
+   * registry,
+   * of the mainnet
+   *
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  mainnet(): this {
+    this.metadataRpc(mainnetMetadataRpc);
+    this.buildRegistry();
+    return this;
   }
 
-  // region Getters and Setters
   /** @inheritdoc */
   protected get transaction(): Transaction {
     return this._transaction;
   }
-  // endregion
 
-  // region Validators
```

Comment:
Leave the regions and organize the methods accordingly, it helps organize the code